homeassistant/components/sun/__init__.py

```diff
@@ -164,7 +165,7 @@ def update_events(self, utc_point_in_time):
         )
         self.location.solar_depression = -10
         self._check_event(utc_point_in_time, "dawn", PHASE_SMALL_DAY)
-        self.next_noon = self._check_event(utc_point_in_time, "solar_noon", None)
+        self.next_noon = self._check_event(utc_point_in_time, "noon", None)
```

Comment:
are `solar_noon` and `noon` equivalent?

Reply:
I don't think so. Solar noon is when the sun is highest in the sky. Noon is 12:00.

Reply:
The docstring for the noon function indicates: `Calculates the solar noon (the time when the sun is at its highest point.)` and reverse for midnight.

Reply:
`noon` and `midnight` were renamed here:
https://github.com/sffjunkie/astral/commit/7e061884b72d3e4c77c34c665607db2be3ed4612